src/include/migraphx/raw_data.hpp

```diff
@@ -191,6 +191,14 @@ struct raw_data : raw_data_base
         ss << static_cast<const Derived&>(*this);
         return ss.str();
     }
+
+    template <class T>
+    std::vector<T> to_vector() const
+    {
+        std::vector<T> result(static_cast<const Derived&>(*this).get_shape().elements());
+        this->visit([&](auto x) { result.assign(x.begin(), x.end()); });
```

Comment:
This should assert that it is not of type "tuple".

Reply:
We can make use of this in many of our tests which currently uses visit.  It could be a good first time issue. Can you open an issue and tag as "good first issue" ?

Reply:
The `visit` already throws an error on tuple type.